include/circt/Dialect/HW/HWAggregates.td

```diff
@@ -139,6 +139,23 @@ def ArrayGetOp : HWOp<"array_get",
   let hasFolder = 1;
 }
 
+def ArrayInjectOp : HWOp<"array_inject",
+    [NoSideEffect,ArrayElementTypeConstraint<"newValue", "input">,
+     TypesMatchWith<"input and result types should be equivalent",
+                     "input", "result", [{ $_self }]>]> {
+  let summary = "Inject a value at the given index in an array";
```

Comment:
Please add a description with a little more detail. It's helpful to link to the website later. For example, the commit message indicates that this overwrites the element at `index`, as opposed to bumping elements to the right like c++ std::vector::insert, or python list.insert. This is where my mind went and I had to go re-read the commit message.

Reply:
Yeah, I also don't like the name of `StructInjectOp`.

Reply:
I haven't thought about it too much, but at least "inject" reserves "insert" for a version that works like other language's "insert" 🤷

Reply:
I would prefer `replace` for both.

Reply:
We can address this consistently at a later point, when both ops have been implemented already.